### ISSUE TITLE
timestep.cpp(85): error: Implicit capture of 'this' in extended lambda expression

### ISSUE THREAD

**Issue by bcfriesen** [AMReX-Astro/Castro#1290]
When compiling Castro commit dc1edf2fd with HPC SDK 20.9 and CUDA 11.1.1, one encounters the following error when compiling `wdmerger` with `make COMP=hpcsdk USE_MPI=FALSE USE_CUDA=TRUE`:

```
Compiling timestep.cpp ...
nvcc -ccbin=g++ -Xcompiler=' -Werror=return-type -g -O3 -std=c++14  -pthread -std=c++14' --std=c++14 -Wno-deprecated-gpu-targets -m64 -arch=compute_70 -code=sm_70 -maxrregcount=255 --expt-relaxed-constexpr --expt-extended-lambda -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -lineinfo --ptxas-options=-O3 --use_fast_math  --forward-unknown-to-host-compiler --Werror ext-lambda-captures-this -dc -x cu   -DEXTRA_THERMO -DCUDA -DGPU_COMPATIBLE_PROBLEM -DAMREX_GIT_VERSION=\"20.11-dirty\" -DAMREX_USE_CUDA_FORTRAN -DAMREX_USE_CUDA -DAMREX_USE_GPU_PRAGMA -DAMREX_LAUNCH='attributes(global)' -DAMREX_DEVICE='attributes(device)' -DAMREX_CUDA_FORT_GLOBAL='attributes(global)' -DAMREX_CUDA_FORT_DEVICE='attributes(device)' -DAMREX_CUDA_FORT_HOST='attributes(host)' -DAMREX_CUDA_FORT_HOST_DEVICE='attributes(host) attributes(device)' -DAMREX_USE_GPU -DBL_COALESCE_FABS -DBL_SPACEDIM=3 -DAMREX_SPACEDIM=3 -DBL_FORT_USE_UNDERSCORE -DAMREX_FORT_USE_UNDERSCORE -DBL_Linux -DAMREX_Linux -DAMREX_DIMENSION_AGNOSTIC -DNDEBUG -DAMREX_GPUS_PER_SOCKET=4 -DAMREX_GPUS_PER_NODE=8 -DAMREX_GPU_MAX_THREADS=256 -DCRSEGRNDOMP -DSPONGE -DGRAVITY -DREACTIONS -DHYBRID_MOMENTUM -DROTATION -DCASTRO_MLMG -DNAUX_NET=0 -DINTEGRATOR=0 -DVODE -DBS -DNUMSCREEN=24 -Itmp_build_dir/s/3d.hpcsdk.CUDA.EXE -I. -I/usr/common/software/sles15_cgpu/cuda/11.1.1/include -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/amrex/Src/Base -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/amrex/Src/AmrCore -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/amrex/Src/Amr -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/amrex/Src/Boundary -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/amrex/Src/LinearSolvers/MLMG -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/util -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/VODE -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/VODE/cuVODE/source/ -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/BS -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/utils -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/rates -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/screening -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/neutrinos -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Exec/science/wdmerger -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/driver -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/hydro -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/problems -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/sources -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Util/model_parser -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Util/interpolate -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/hydro -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/gravity -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/scf -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/reactions -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/Source/rotation -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/EOS -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/EOS/helmholtz -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/networks/aprox13 -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/EOS -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/networks -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/interfaces -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/constants -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/VODE -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/VODE/cuVODE/source/ -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/BS -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration/utils -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/integration -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/rates -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/screening -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/Microphysics/neutrinos -Itmp_build_dir/castro_sources/3d.hpcsdk.CUDA.EXE -I/global/homes/f/friesen/NESAP/AMReX/Castro.git/external/amrex/Tools/C_scripts -c tmp_build_dir/s/3d.hpcsdk.CUDA.EXE/timestep.cpp -o tmp_build_dir/o/3d.hpcsdk.CUDA.EXE/timestep.o
tmp_build_dir/s/3d.hpcsdk.CUDA.EXE/timestep.cpp(85): error: Implicit capture of 'this' in extended lambda expression
```

The code in question is this:

```
 78 #ifdef ROTATION
 79       if (ldo_rotation == 1 && lstate_in_rotating_frame != 1) {
 80         Real vel[3];
 81         vel[0] = ux;
 82         vel[1] = uy;
 83         vel[2] = uz;
 84 
 85         GeometryData geomdata = geom.data();
 86 
 87         inertial_to_rotational_velocity_c(i, j, k, geomdata,
 88                                           omega.begin(), time, vel);
 89 
 90         ux = vel[0];
 91         uy = vel[1];
 92         uz = vel[2];
 93       }
 94 #endif
```

**Comment by maxpkatz** [comment]
I'll fix this.

**Comment by maxpkatz** [comment]
@bcfriesen I believe this was fixed in #1291, can you please try again?

**Comment by bcfriesen** [comment]
It compiles without error now, thanks! I will resolve this.
